### ISSUE TITLE
Cannot cleanly exit chroot

### ISSUE THREAD

**Issue by TheChymera** [IBT-FMI/gebuilder#22]
The [`gebuild_enter`](https://github.com/IBT-FMI/gebuilder/blob/master/gebuilder/gebuild_enter) and [`gebuild_exit`](https://github.com/IBT-FMI/gebuilder/blob/master/gebuilder/gebuild_exit) functions are intended to ease the chrooting into and out of failed initializations.

Sadly the chrooting our of, seems to not work that cleanly, e.g.:

```
newbuilder /usr/share/gebuilder/roots # gebuild_exit e6966aabd76aeb9189f3f03a02c59681a37ffdbe2d89c8d842a071807422ccda/root/
umount: /usr/share/gebuilder/roots/e6966aabd76aeb9189f3f03a02c59681a37ffdbe2d89c8d842a071807422ccda/root/dev: target is busy.
umount: e6966aabd76aeb9189f3f03a02c59681a37ffdbe2d89c8d842a071807422ccda/root//proc: not mounted.
umount: e6966aabd76aeb9189f3f03a02c59681a37ffdbe2d89c8d842a071807422ccda/root//tmp: not mounted.
umount: /usr/share/gebuilder/roots/e6966aabd76aeb9189f3f03a02c59681a37ffdbe2d89c8d842a071807422ccda/root/sys: target is busy.
```

This is ofc after having run `exit` inside the chroot.
The only solution atm seems to be rebooting...

@Doeme any ideas what could be wrong?

**Comment by Doeme** [comment]
Yes. `mount --rbind` recurses into mountpoints inside mountpoints, i.e. those have to be unmounted recursively too, with `umount -R`

**Comment by Doeme** [comment]
Also, I'm not sure why you do not use the `./gebuilder <.gentoo> shell` function

**Comment by TheChymera** [comment]
@Doeme Lol, because I was not aware of it.

Do we have an unmounting script to go along with `./gebuilder <.gentoo> shell`?

Also, neither `gebuild_enter` nor `./gebuilder <.gentoo> shell` seem to update the shell prompt so that [it's obvious that it's a chroot](https://github.com/IBT-FMI/gebuilder/blob/777ba9ae57d6075f448cc70bd54d360ac4c9d474/gebuilder/gebuild_enter#L12).
Any idea how so?

**Comment by Doeme** [comment]
The shell-action doesn't do this because I personally never used it. It can be added, though.
`gebuild_enter` does not do this, because PS1 is overridden once `/etc/bash/bashrc` is loaded.
